### PR TITLE
Author for articles

### PR DESCRIPTION
**Pull request by IgnacioColuccio**
### Summary
- Added user field for commentaries
- Added access control for articles (can't edit unless it's your own post)
Note: Specs are missing, assign this back to me if you want the relationship to be tested.

### Trello card
https://trello.com/c/0Mvva2Lb/29-agregarle-creador-a-los-articulos-y-comentarios

### Test plan
1. Create a post while logged out.
2. It should redirect to the sign up page.
3. After signing up or signing in, you should be able to create posts.
4. Login as someone else and create a post.
5. It should be possible to edit your own posts but not the one from the first user.

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -28,6 +26,7 @@ def new
 
   def create
     @article = Article.new(article_params)
+    @article.user = current_user
```
Comment (Dinuuu) [inline app/controllers/articles_controller.rb]: you can use
```ruby
@article = Article.new(article_params.merge(user: current_user))
```

File: app/models/article.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class Article < ActiveRecord::Base
+  belongs_to :user
```
Comment (Dinuuu) [inline app/models/article.rb]: you should also validate that this user is present in the article

File: app/models/user.rb [hunk 3]
```diff
@@ -3,4 +3,5 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  has_many :articles
```
Comment (Dinuuu) [inline app/models/user.rb]: What happen if a User is destroyed ? His articles should be destroyed to, for that use `dependent: :destroy`

**Comment by IgnacioColuccio** [comment]
Fixed those things

**Comment by Dinuuu** [comment]
Rebase it and :+1:

**Pull request merged**